# Lab 1: Intro Python Practice (5 points)

Instructions:
- Feel free to experiment by modifying the code and re-running the cell.
- Successful submission: 1 point
- Each question: 1 point (4x1)
- To get the point, you need to complete the code, give the answer, and provide explanation (only if required).


## Question 1: Area of a circle

Compute the area of a circle with radius `r = 5` using the formula `A = π r^2`.


In [3]:
import math
r = 5
area = math.pi * (r ** 2)
print(f"{area:.2f}")


78.54


Your answer: 78.54


## Question 2: Sum of even numbers

Given the list `nums = [1,2,3,4,5,6,7,8,9,10]`, compute the sum of the even numbers.


In [64]:
nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
total_evens = 0
for i in range(len(nums)):
    if nums[i] % 2 == 0:
        total_evens = total_evens + nums[i]

print(total_evens)


30


This loop will enumerate through the `nums` list by checking its total length of the list. Then we check if the value at our index value of `nums` mod 2 has no remainder. If It does have no remainder we add our previous result into our variable `total evens` and the even index. Finally we print out `total evens`.

## Question 3: Count vowels in a string

Count the number of vowels in the string `s = "Hello, world!"` (consider both uppercase and lowercase).


In [14]:
s = "Hello, world!"
vowels = set("aeiouAEIOU")

count = 0

for i in range(len(s)):
    if s[i] in vowels:
        count = count + 1

print(count)


3


I first create a variable to keep track of our count of vowels named `count`. Then we head into a loop that enumerates the total length of our string `s` then we check to see if the index value of our string `s` is anywhere in the object `vowels`. If it is then we increase the value of our `count` object which is then printed after the loop is finished.


## Question 4: Factorial function

Write a function to compute the factorial of `n` (e.g., `factorial(5) == 120`) using an iterative approach.


In [69]:
def factorial(n):
    if type(n) == int and n > 0:
        for i in range (1, n):
            n = n * i
        return n

    else:
        print("Invalid type or value for n")

factorial(5)

120

The function `factorial` takes in an interger parameter `n`, if an integer is not provided or it is not greater than zero then we print out an error message. If we are provided an integer we enter a loop that starts from `1` and iterates until `n`. We start from `1` because if we didn't we would be multiplying by zero defeating the purpose of factorial. Once we are in the loop the function assigns the value of n as itself times the current value of the index `i`. We return `n` once the function is has reached the value of `n`.